### ISSUE TITLE
Cannot use resets in a circuit

### ISSUE THREAD

**Issue by nonhermitian** [Qiskit/qiskit-terra#5736]
### Information

- **Qiskit Terra version**: latest
- **Python version**:
- **Operating system**:

### What is the current behavior?
```
     ┌─────────┐     ┌─┐
q_0: ┤ unitary ├─|0>─┤M├
     └─────────┘     └╥┘
c: 1/═════════════════╩═
                      0 
```
gives:
```
~/miniconda3/envs/qiskit/lib/python3.7/site-packages/qiskit/quantum_info/operators/operator.py in _append_instruction(self, obj, qargs)
    526             # cannot compose this gate and raise an error.
    527             if obj.definition is None:
--> 528                 raise QiskitError('Cannot apply Instruction: {}'.format(obj.name))
    529             if not isinstance(obj.definition, QuantumCircuit):
    530                 raise QiskitError('Instruction "{}" '

QiskitError: 'Cannot apply Instruction: reset'
```

even if the backend supports it.  The code I am using used to work in previous versions.

**Comment by mtreinish** [comment]
What operations are you running on that circuit to raise the error? That exception is being raised from trying to create an operator from something (I assume the circuit given it's erroring on reset). We probably just need to add special handling to the operator constructor to deal with reset instructions.

As far as I know everything in the circuit -> transpile -> assemble -> execute path works for reset (although there are some complaints about resets being removed during transpilation #5127, although that case isn't a hard failure it just optimizes away resets during a transpile)

**Comment by nonhermitian** [comment]
It is a random 1Q unitary followed by a reset and meas (as the circuit shows), then passed to device using `execute`.  I can confirm the exact same code works on `0.23.2`.

**Comment by mtreinish** [comment]
Ok, that's what I needed to know, it's being raised by a transpile (that wasn't in the orriginal issue). This is actually a duplicate of #5543 (although it should be fixed by #5570) and we unfortunately backported a regression in 0.16.2 with #5655 (which I thought should have fixed it). Which means we'll need to fix it for real on the stable branch and push out a terra 0.16.3.

**Comment by mtreinish** [comment]
FWIW, this should work fine on master, it appears to just be a bug in that backport :(

**Comment by nonhermitian** [comment]
Ok cool.  Getting a fix out would be nice since reset is a new(ish) feature we would like to highlight.

**Comment by mtreinish** [comment]
Yeah, I've got the patch ready, just adding a test now. As soon as that merges to stable/0.16 we'll prepare and push out a 0.16.3 release
